src/main/java/jenkins/plugins/git/GitToolChooser.java

```diff
@@ -126,35 +136,76 @@ private boolean setSizeFromAPI(String repoUrl, Item context, String credentialsI
      * @param sizeOfRepo: Size of a repository (in KiBs)
      * @return a git implementation, "git" or "jgit"
      */
-    String determineSwitchOnSize(Long sizeOfRepo, String gitExe) {
+    String determineSwitchOnSize(Long sizeOfRepo, GitTool tool) {
         if (sizeOfRepo != 0L) {
             if (sizeOfRepo < SIZE_TO_SWITCH) {
                 if (!JGIT_SUPPORTED) {
                     return "NONE";
                 }
-                return determineToolName(gitExe, JGitTool.MAGIC_EXENAME);
+                GitTool rTool = resolveGitToolForRecommendation(tool, JGitTool.MAGIC_EXENAME);
+                if (rTool == null) {
+                    return "NONE";
+                }
+                return rTool.getGitExe();
             } else {
-                return determineToolName(gitExe, "git");
+                GitTool rTool = resolveGitToolForRecommendation(tool, "git");
+                if (rTool == null) {
+                    return "NONE";
```

Comment:
This statement is never reached by the tests in GitToolChooserTest.  Could it be reached reasonably by an addition of one of those tests?

Reply:
Thanks for catching this. It can't be reached because there is no possible scenario where the tool will be null. I have removed it.